cwl/src/main/scala/cwl/CommandLineTool.scala

```diff
@@ -49,25 +49,30 @@ case class CommandLineTool private(
   private [cwl] implicit val explicitWorkflowName = ParentName(id)
   private val inputNames = this.inputs.map(i => FullyQualifiedName(i.id).id).toSet
 
+  // Circe can't create bidirectional links between workflow steps and runs (including `CommandLineTool`s) so this
+  // ugly var is here to link back to a possible parent workflow step. This is needed to navigate upward for finding
+  // requirements in the containment hierarchy. There isn't always a containing workflow step so this is an `Option`.
+  private[cwl] var parentWorkflowStep: Option[WorkflowStep] = None
```

Comment:
Does this work even in a multiple file scenario? (I notice your test case is an all-in-one cwl file)

Reply:
Doesn't SALAD all-in-one everything before we begin to parse?